gateway/src/main/java/com/lyft/data/gateway/router/RoutingManager.java

```diff
@@ -36,17 +36,22 @@
   private final AtomicLong requestAdhocCounter = new AtomicLong(0);
   private final AtomicLong requestScheduledCounter = new AtomicLong(0);
   private final Cache<String, String> queryIdBackendCache;
+  private final Cache<String, String> transactionIdBackendCache;
   private ExecutorService executorService = Executors.newFixedThreadPool(5);
   private GatewayBackendManager gatewayBackendManager;
 
   public RoutingManager(GatewayBackendManager gatewayBackendManager, String cacheDataDir) {
     this.gatewayBackendManager = gatewayBackendManager;
+    this.queryIdBackendCache = buildCache(cacheDataDir, "queryIdBackendCache");
+    this.transactionIdBackendCache = buildCache(cacheDataDir, "transactionIdBackendCache");
```

Comment:
Do you think it'd make sense to set an expiration (based on tx timeout value) for this cache? Also, can the entry be removed from the cache on commit/rollback?

Reply:
We would need to explore more on how long a `transaction id` is valid, I have not checked the presto-code, but I guess expiration would be refreshed as client sends more queries. Also this cache only allocates `100mb` in-memory, after that it starts pushing stale values to disc which has 1GB limit, so we are expiring all the ids beyond that limit.